source/gui/addonGui.py

```diff
@@ -248,7 +271,23 @@ def onHelp(self, evt):
 		path = self.curAddons[index].getDocFilePath()
 		os.startfile(path)
 
-	def OnGetAddonsClick(self,evt):
+	def onToggle(self, evt):
+		index=self.addonsList.GetFirstSelected()
+		if index<0: return
+		addon=self.curAddons[index]
+		shouldDisable = self._shouldDisable(addon)
+		if shouldDisable:
+			# Translators: Presented when attempting to disable the selected add-on.
+			if gui.messageBox(_("You are about to disable the selected add-on. If you need to use features provided by this add-on, you need to enable this add-on from add-ons manager. Are you sure you wish to continue and disable this add-on?"),

```

Comment:
I felt it best to warn users before letting them confirm, similar to the warning shown when users install new add-ons. But I can tell this could become annoying for some...